### DISCUSSION TITLE
Yarn monorepo - trying to import a js file with export statements says export is reserved

### DISCUSSION THREAD

**Discussion by Madd0g** [quasarframework/quasar#15941] [General - Components / Directives / etc]
I have a bunch of packages in my monorepo - I have a shared package that is shared between client and server. In the shared package I have both `exports` and `types` in package.json.

When I try to import that in my quasar project, I get the error that "export is a reserved keyword".

Does that mean that for quasar I need to package as commonjs or something? I tried that and it breaks my server, which uses that package without issue.

Any advice for me? I saw lots of discussions about supporting ESM in the quasar config, but couldn't find people talking about these problems with dependencies.

EDIT: found the answer here for [bundling linked dependencies in the vite docs site](https://vitejs.dev/guide/dep-pre-bundling.html#monorepos-and-linked-dependencies)

Thanks

**Comment by lugrinder** [comment]
@Madd0g Hi, can you explain how you solved this? I've followed the vide docs about bundling linked dependencies, but still fails, probably I don't understand how to set up.

**Reply by Madd0g** [reply]
sorry, it's been too long, I don't have many quasar projects, but I searched them and I couldn't find this code anywhere. I must have solved the issue some other way eventually? Don't even know.

**Comment by lugrinder** [comment]
@Madd0g Thanks... I'll wait to see if someone reads this post and enlightens me...

**Comment by lugrinder** [comment]
I've solved the problem: adding the backend (the shared package) scope path to the `build.alias` of `quasar.conf.js`.

For example, if the backend (or shared library) is named `@my-scope/backend` in the `package.json`, add that package name as a dependency in the frontend (Quasar):

```json
dependencies: {
    "@my-scope/backend": "workspace:*"
}
```

And add the scope to the `build.alias` of `quasar.conf.js`:

```javascript
build: {
    alias: {
        '@my-scope': path.resolve(__dirname, '../')
    }
}
```

In this case, `resolve` is considering that both, backend and frontend (Quasar) packages are on the same folder.

To import any file from the backend in the Quasar, use scope:

```typescript
import type { MyType } from '@my-scope/backend/src/myfile'
```
